input/system/azure/logs.go

```diff
@@ -229,20 +240,20 @@ func SetupLogSubscriber(ctx context.Context, wg *sync.WaitGroup, globalCollectio
 }
 
 func GetServerNameFromRecord(in AzurePostgresLogRecord) string {
-	if in.LogicalServerName == "" { // Flexible Server
-		// For Flexible Server, logical server name is not set, so instead determine it based on the resource ID
+	if in.IsSingleServer() {
+		return in.LogicalServerName
+	} else { // Flexible Server, Cosmos DB
+		// For Flexible Server, logical server name is typically not set, so instead determine it based on the resource ID
 		resourceParts := strings.Split(in.ResourceID, "/")
 		return strings.ToLower(resourceParts[len(resourceParts)-1])
-	} else { // Single Server
-		return in.LogicalServerName
 	}
 }
 
 // Parses one Azure Event Hub log record into one or two log lines (main + DETAIL)
 func ParseRecordToLogLines(in AzurePostgresLogRecord, parser state.LogParser) ([]state.LogLine, error) {
 	logLineContent := in.Properties.Message
 
-	if in.LogicalServerName != "" { // Single Server
+	if in.IsSingleServer() { // Single Server
```

Comment:
Now that you have helper methods here, the comments are superfluous.
```suggestion
	if in.IsSingleServer() {
```